ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/StreamQueryResponseHandler.java

```diff
@@ -49,52 +54,72 @@
   }
 
   @Override
-  protected void handleMetadata(final QueryResponseMetadata queryResponseMetadata) {
+  protected void handleMetadata(final StreamedRow queryResponseMetadata) {
+    final String queryId = queryResponseMetadata.getHeader().get().getQueryId().toString();
+
+    final List<Column> columns = queryResponseMetadata
+            .getHeader()
+            .get()
+            .getSchema()
+            .columns();
+
+    final List<String> columnNames = columns
+            .stream()
+            .map(col -> col.name().text())
+            .collect(Collectors.toList());
+
+    final List<String> columnTypes = columns
+            .stream()
+            .map(col -> col.type().toString())
+            .collect(Collectors.toList());
+
     this.queryResult = new StreamedQueryResultImpl(
         context,
-        queryResponseMetadata.queryId,
-        queryResponseMetadata.columnNames,
-        RowUtil.columnTypesFromStrings(queryResponseMetadata.columnTypes)
+        queryId,
+        columnNames,
+        RowUtil.columnTypesFromStrings(columnTypes)
     );
-    this.columnNameToIndex = RowUtil.valueToIndexMap(queryResponseMetadata.columnNames);
+
+    this.columnNameToIndex = RowUtil.valueToIndexMap(columnNames);
     cf.complete(queryResult);
   }
 
   @Override
-  protected void handleRow(final Buffer buff) {
+  protected void handleRow(final StreamedRow row) {
     if (queryResult == null) {
       throw new IllegalStateException("handleRow called before metadata processed");
     }
-    final Object json = buff.toJson();
-    final Row row;
-    if (json instanceof JsonArray) {
-      row = new RowImpl(
+
+    if (row.getConsistencyToken().isPresent()) {
+      // This is the serialized consistency vector
+      // Don't add it to the publisher's buffer since the user should not see it
+      final ConsistencyToken consistencyToken = row.getConsistencyToken().get();
+      try {
+        final String consistencyTokenString = consistencyToken.getConsistencyToken();
+        LOG.info("Response contains consistency vector " + consistencyTokenString);
+        serializedConsistencyVector.set(consistencyTokenString);
+      } catch (Exception e) {
```

Comment:
Same here

Reply:
yes, you are right. removed try...catch block